src/display/api.js

```diff
@@ -2550,6 +2558,18 @@ class WorkerTransport {
       });
   }
 
+  getAnnotationObjects() {
+    return this.messageHandler
+      .sendWithPromise("GetAnnotationObjects", {
+        numPages: this._numPages,
+      })
+      .then(data => {
+        return data
+          .reduce((acc, x) => acc.concat(x), [])
+          .filter(x => x !== null);
```

Comment:
This feels overly terse, and it's honesty a bit difficult to understand exactly what this code actually accomplishes.
Can this be improved somewhat? And please also use longer/more descriptive variable names here.

Finally, any reason not to do these manipulations on the worker-side before sending the data to the API?